### DISCUSSION TITLE
fastapi type stubs

### DISCUSSION THREAD

**Discussion by tekumara** [fastapi/fastapi#7407] [Questions] [answered]
Would be nice if fastapi shipped with `.pyi` type stubs so I can type check my application using pyright.

**Comment by kajogo777** [comment]
yes, this would be great!!

**Comment by tiangolo** [comment, marked as answer]
Hey there, FastAPI already has the source code fully annotated and the metadata to mark it like that. `.pyi` files are only when the code is not annotated. You should be able to check your code with Pyright or mypy.

For reference, directly from Pyright: https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#inlined-type-annotations-and-type-stubs
